programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1139,6 +1139,14 @@ declare_builtin_function!(
         use solana_zk_token_sdk::curve25519::{
             curve_syscall_traits::*, edwards, ristretto, scalar,
         };
+
+        let restrict_msm_length = invoke_context
+            .feature_set
+            .is_active(&feature_set::curve25519_restrict_msm_length::id());
+        if restrict_msm_length && points_len > 512 {
+            return Err(Box::new(SyscallError::InvalidLength));
+        }
```

Comment:
```suggestion
        #[allow(clippy::collapsible_if)]
        if restrict_msm_length {
            if points_len > 512 {
                return Err(Box::new(SyscallError::InvalidLength));
            }
        }
```

makes the removal pr cleaner

Reply:
yes, good point!